linode/manager
# change: [M3-8390] - Initialize Pendo on Cloud Manager

## Description 📝
This PR adds Pendo to Cloud Manager by adding the Pendo agent script from the CDN and initializing Pendo with some custom configurations.

## Changes  🔄
- Installs `js-sha256`
- Creates the `usePendo` hook
   - Incorporates Pendo's install script code
   - Loads the Pendo Agent from the CDN with `loadScript`
   - Initializes Pendo with some custom configuration
- Invokes the hook in `App.tsx` with the rest of the GlobalListeners

## Target release date 🗓️
10/14/24

## How to test 🧪

### Prerequisites
(How to setup test environment)
- To your `.env`, add the Pendo API Key. This is not a secret key; it will be visible in the browser dev tools, but we also want to be able to stop Pendo from running using an environment variable, which will live elsewhere with our env var configurations. See the internal repo (PR 167) for that change and the API key.
```
# Pendo:
REACT_APP_PENDO_API_KEY='KEY-GOES-HERE'
```

### Verification steps
(How to verify changes)
- We can validate that Pendo is installed by opening the browser console and typing `pendo.validateEnvironment()`. This should output some data, including your hashed visitor and account id.
- We also want to confirm that no sensitive data is sent to Pendo. To do this, filter by `pendo` in the Network tab of the browser dev tools.
   - Then try navigating around Cloud Manager and confirm the following by looking at the `normalizedUrl` of the requests:
      - No ids are being sent (e.g. when viewing a linode details page, `linodes/123` should be truncated to `linodes/`)
      - No search queries are being sent
      - No username or other account data is being sent (e.g. when visiting the User Profile and User Permissions pages, the url doesn't include the username)
      - No token data is being sent

Example:
![image](https://github.com/user-attachments/assets/f03dc759-9a1e-486c-84fc-f36d47dd6789)


## As an Author I have considered 🤔

*Check all that apply*
